src/zabapgit_object_form.prog.abap

```diff
@@ -0,0 +1,330 @@
+      TABLES
+        using_tab = lt_bdcdata
+      EXCEPTIONS
+        OTHERS    = 1
+        ##fm_subrc_ok.                                                   "#EC CI_SUBRC
+
+  ENDMETHOD.
+
+  METHOD lif_object~delete.
+
+    DATA: lv_name TYPE thead-tdname.
+
+    lv_name = ms_item-obj_name.
+
+    CALL FUNCTION 'DELETE_TEXT'
+      EXPORTING
+        id       = 'TXT'
+        language = '*'
+        name     = lv_name
+        object   = c_objectname_form
+      EXCEPTIONS
+        OTHERS   = 1.
+    IF sy-subrc <> 0.
+      lcx_exception=>raise( 'error from DELETE_TEXT TXT, FORM' ) ##NO_TEXT.
+    ENDIF.
+
+    lv_name+16(3) = '*  '.
+    CALL FUNCTION 'DELETE_TEXT'
+      EXPORTING
+        id       = c_tdid_default
+        language = '*'
+        name     = lv_name
+        object   = c_objectname_form
+      EXCEPTIONS
+        OTHERS   = 1.
+    IF sy-subrc <> 0.
+      lcx_exception=>raise( 'error from DELETE_TEXT DEF, FORM' ) ##NO_TEXT.
+    ENDIF.
+
+  ENDMETHOD.
+
+  METHOD lif_object~serialize.
+
+    DATA: lt_form              TYPE tyt_header.
+    DATA: ls_header            TYPE LINE OF tyt_header.
+    DATA: lt_lines             TYPE tyt_lines.
+    DATA: lv_name              TYPE thead-tdname.
+    DATA: lt_header            TYPE tyt_header.
+    DATA: lv_string            TYPE string.
+    DATA: lo_xml               TYPE REF TO lcl_xml_output.
+    FIELD-SYMBOLS: <ls_header> TYPE LINE OF tyt_header.
+
+    lv_name = ms_item-obj_name.
+
+    CALL FUNCTION 'SELECT_TEXT'
+      EXPORTING
+        database_only = abap_true
+        id            = '*'
+        language      = '*'
+        name          = lv_name
+        object        = c_objectname_form
+      TABLES
+        selections    = lt_header
+      EXCEPTIONS
+        OTHERS        = 1
+        ##fm_subrc_ok.                                                   "#EC CI_SUBRC
+
+    LOOP AT lt_header ASSIGNING <ls_header>.
+      CLEAR ls_header.
+      CLEAR lt_lines.
+      FREE lo_xml.
+
+      CALL FUNCTION 'READ_TEXT'
+        EXPORTING
+          id       = <ls_header>-tdid
+          language = <ls_header>-tdspras
+          name     = <ls_header>-tdname
+          object   = <ls_header>-tdobject
+        IMPORTING
+          header   = ls_header
+        TABLES
+          lines    = lt_lines
+        EXCEPTIONS
+          OTHERS   = 1
+          ##fm_subrc_ok.                                                   "#EC CI_SUBRC
+
+      CLEAR: ls_header-tdfuser,
+             ls_header-tdfdate,
+             ls_header-tdftime,
+             ls_header-tdfreles,
+             ls_header-tdluser,
+             ls_header-tdldate,
+             ls_header-tdltime,
+             ls_header-tdlreles.
+
+      CREATE OBJECT lo_xml.
+      lo_xml->add( iv_name = 'TLINES'
+                   ig_data = lt_lines ).
+      lv_string = lo_xml->render( ).
+      IF lv_string IS NOT INITIAL.
+        mo_files->add_string( iv_extra  = _build_extr_from_header( ls_header )
```

Comment:
if you put everything into a abap structure, then the data can be added/read in one go with `lo_xml->add`, I dont think multiple files are required in this case